### PR TITLE
HPCC-29831 Allow localAgent setting to be set in workunit/regression suite

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Via regression suite.

### CODE REVIEW

File: testing/regress/ecl/stresstext.ecl [hunk 1]
```diff
@@ -23,6 +23,10 @@
 //version multiPart=false,variant='inplace_lzw'
 //version multiPart=false,variant='inplace_lz4hc'
 
+#option('roxie:minPayloadSize', 10000)
```
Comment (ghalliday) [inline testing/regress/ecl/stresstext.ecl:26]: I don't think these changes should be released.

File: common/workunit/workunit.cpp [hunk 2]
```diff
@@ -8458,7 +8458,9 @@ void CLocalWorkUnit::copyWorkUnit(IConstWorkUnit *cached, bool copyStats, bool a
 bool CLocalWorkUnit::hasDebugValue(const char *propname) const
 {
     StringBuffer lower;
-    lower.append(propname).toLowerCase();
+    lower.append(propname);
```
Comment (ghalliday) [inline common/workunit/workunit.cpp:8461]: @jakesmith can you think of any problems this would cause?   I haven't found any.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman looks good.  One test case to remove your testing defaults.  Also, should it target 9.2.x rather than master?

File: common/workunit/workunit.cpp (hunk 2 above)
Comment (jakesmith) [reply common/workunit/workunit.cpp:8461]: No, but I might be tempted to put these 3 common lines in a helper method/function, with a comment saying why it is differentiating.

Comment (jakesmith) [reply common/workunit/workunit.cpp:8461]: No, but I might be tempted to put these 3 common lines in a helper method/function, with a comment saying why it is differentiating.

**Review by jakesmith — COMMENTED** [review]
@richardkchapman - 1 additional comment.

File: testing/regress/ecl/stresstext.ecl (hunk 1 above)
Comment (richardkchapman) [reply testing/regress/ecl/stresstext.ecl:26]: I will comment them out.

**Comment by richardkchapman** [comment]
@ghalliday Updated test case and squashed

**Review by ghalliday — APPROVED** [review]
I would also common the 3 lines into a shared function, but happy to merge as-is.

**Pull request merged**